// tododbvi/application.js
// status se TRUE = DA SVOLGERE se FALSE = SVOLTO
exports.todoList = todoList = [{
  idTodo:1,
  idCreatore:3,
  idAssegnatario : 2,
  azioneDaSvolgere: 'comprare il pane',
  completed:false
},
{
  idTodo:2,
  idCreatore:3,
  idAssegnatario : 1,
  azioneDaSvolgere: 'comprare il vino',
  completed:false
},
{
  idTodo:3,
  idCreatore:2,
  idAssegnatario : 3,
  azioneDaSvolgere: 'comprare la pasta',
  completed:false
},
{
  idTodo:4,
  idCreatore:3,
  idAssegnatario : 2,
  azioneDaSvolgere: 'comprare la carne',
  completed:false
},
{
  idTodo:5,
  idCreatore:1,
  idAssegnatario : 3,
  azioneDaSvolgere: 'comprare il mascarpone',
  completed:false
}];

exports.utenti = utenti = [{
  idCliente:1,
  name:'vincenzo'
},
{
  idCliente:2,
  name:'carlo'
},
{
  idCliente:3,
  name:'maria'
},
];


exports.add = function (idCreatore,idAssegnatario,azioneDaSvolgere) {
  cond=false; cond2=false; cond3=false;
  for (var i in utenti){
    if(idCreatore==utenti[i].idCliente){
      cond=true;
    }
    if(idAssegnatario==utenti[i].idCliente){
      cond2=true;
    }
  }
  if(idCreatore==idAssegnatario && idCreatore!=undefined && idAssegnatario!=undefined ){
    cond3=true;
  }

  if(cond3==true){
    message={message:"the id creator and id assignement are equal",status:400};
    return message;
  }

  if(cond==false || cond2==false){
    message={message:"the creator or assignement user in the body does not exist",status:400};
    return message;
  }
  if(azioneDaSvolgere==undefined){
    azioneDaSvolgere="undefined";
  }

  var newID = parseInt(todoList[todoList.length-1].idTodo)+1;

  todoList.push({
    idTodo:newID,
    idCreatore: idCreatore,
    idAssegnatario : idAssegnatario,
    azioneDaSvolgere : azioneDaSvolgere,
    completed:false
  });
  message={message:"the todo was been added",status:200};

  return message;
};

exports.del = function(id){
  id = parseInt(id);
  check=false;
  for(var i=0;i<todoList.length;i++){
    if(id==todoList[i].idTodo){
      todoList.splice(i,1);
      check=true;
      messageDelete={message:"the id was deleted successfully",status:200};
      return messageDelete;
    }
  }
  if(check==false){
    messageDelete={message:"the id is not present in the database",status:400};
    return messageDelete;
  }
};

exports.setCompleted = function(id){
  id = parseInt(id);
  check=false;
  for(var i=0;i<todoList.length;i++){
    if(id==todoList[i].idTodo){
      todoList[i].completed=true;
      check=true;
      messageCompleted={message:"the todo was set completed",status:200};
      return messageCompleted;
    }
  }
  if(check==false){
    messageCompleted={message:"the todo is not present in the database",status:400};
    return messageCompleted;
  }

};

exports.setUncompleted = function(id){
  id = parseInt(id);
  check=false;
  for(var i=0;i<todoList.length;i++){
    if(id==todoList[i].idTodo){
      todoList[i].completed=false;
      check=true;
      messageUncompleted={message:"the todo was set Uncompleted",status:200};
      return messageUncompleted;
    }
  }
  if(check==false){
    messageUncompleted={message:"the todo is not present in the database",status:400};
    return messageUncompleted;
  }

};

exports.showCompletedTodos = function(){
  var todosComplete=[];
  check=false;
  for(var i=0;i<todoList.length;i++){
      if(todoList[i].completed==true){
        todosComplete.push(todoList[i]);
        check=true;
      }
  }

  if(check==false){
    message={message:"no todos completed avable",status:401};
    return message;
  }else{
    return todosComplete;
  }

};

exports.showUncompletedTodos = function(){
  var todosUncomplete=[];
  check=false;
  for(var i=0;i<todoList.length;i++){
      if(todoList[i].completed==false){
        todosUncomplete.push(todoList[i]);
        check=true;
      }
  }

  if(check==false){
    message={message:"no todos UnCompleted avable",status:401};
    return message;
  }else{
    return todosUncomplete;
  }

};

exports.showTodosCreator = function(user){
  var todos=[];
  check=false;
  for(var i=0;i<todoList.length;i++){
    if(todoList[i].idCreatore==user){
      todos.push(todoList[i]);
      check=true;
    }
  }

  if(check==false){
    message={message:"no todos whit this id",status:401};
    return message;
  }else{
    return todos;
  }

};

exports.showTodosAssignee = function(user){
  var todos=[];
  check=false;
  for(var i=0;i<todoList.length;i++){
    if(todoList[i].idAssegnatario==user){
      todos.push(todoList[i]);
      check=true;
    }
  }

  if(check==false){
    message={message:"no todos whit this id",status:401};
    return message;
  }else{
    return todos;
  }

};
